intel_extension_for_transformers/llm/runtime/graph/core/ne_layers.c

```diff
@@ -7334,6 +7355,20 @@ static void ne_compute_forward_diag_mask_f32(const struct ne_compute_params* par
   assert(dst->nb[0] == sizeof(float));
   assert(src0->nb[0] == sizeof(float));
 
+  // mask padding token (padding left)
+  for (int b = 0; b < bs; b++) {
+    const int n_padding = ((int32_t*)src1->data)[2 + b];
+    if (n_padding == 0) continue;
+    for (int k = 0; k < (nz / bs); k++) {
+      for (int j = ith; j < nr; j += nth) {
+        for (int i = 0; i < n_padding; i++) {
+          // if (j + n_past > n_padding - 1)  // it will not affect next token if don't mask the pad_token row
+          *(float*)((char*)dst->data + b * dst->nb[3] + k * dst->nb[2] + j * dst->nb[1] + i * dst->nb[0]) = value;
```

Comment:
I think you can easily vectorize these manually or at least ```pragma omp simd```

Reply:
fixed in [8ba8390](https://github.com/intel/intel-extension-for-transformers/pull/583/commits/8ba83901db1fc87880050fae145e83c160258631)

Reply:
You can use below code to vectorize them.
```
      for (int j = ith; j < nr; j += nth) 

	ne_vec_set_f32(n_padding,(float*)((char*)dst->data + b * dst->nb[3] + k * dst->nb[2] + j * dst->nb[1]),value);
```

Reply:
thanks, updated in [f8c62f5](https://github.com/intel/intel-extension-for-transformers/pull/583/commits/f8c62f5623d571640da1bfd20d9927a4d0eaa9e6)